### ISSUE TITLE
Some questions about the loss function

### ISSUE THREAD

**Issue by HyuanTan** [NVlabs/geomapnet#2]
Hi, thanks for share!
I want to ask some questions about the loss function `MapNetOnlineCriterion`. When calculate the VO loss in training `MapNet++`, the VO loss equals the position error(Just x and y, why exclude the z?):
```
   pred_vos = pred_vos.view(-1, *s[2:])
    targ_vos = targ_vos.view(-1, *s[2:])
    idx = 2 if self.gps_mode else 3
    vo_loss = \
      torch.exp(-self.srx) * self.t_loss_fn(pred_vos[:, :idx], targ_vos[:, :idx]) + \
      self.srx
    if not self.gps_mode:
      vo_loss += \
        torch.exp(-self.srq) * self.q_loss_fn(pred_vos[:, 3:], targ_vos[:, 3:]) + \
        self.srq
```

And when test, the VO loss equals the position error(x,y,z) and orientation error, what make these differences?
```
    if args.model.find('++') >= 0:
    kwargs = dict(kwargs, gps_mode=(vo_lib=='gps') )
    train_criterion = MapNetOnlineCriterion(**kwargs)
    val_criterion = MapNetOnlineCriterion()
```

And in:
```
    # get the VOs
    if not self.gps_mode:
      pred_vos = pose_utils.calc_vos(pred_vos)
```
Why just `pred_vos` using pose_utils.calc_vos(), how about `targ_vos`?

Thanks!!
